tests/kafkatest/tests/streams/streams_standby_replica_test.py

```diff
@@ -0,0 +1,159 @@
+    """
+
+    input_topic = "standbyTaskSource1"
+    output_topic_1 = "standbyTaskSink1"
+    output_topic_2 = "standbyTaskSink2"
+
+    num_messages = 60000
+
+    def __init__(self, test_context):
+        super(StreamsStandbyTask, self).__init__(test_context=test_context)
+        self.zk = ZookeeperService(test_context, num_nodes=1)
+        self.kafka = KafkaService(test_context,
+                                  num_nodes=3,
+                                  zk=self.zk,
+                                  topics={
+                                      self.input_topic: {'partitions': 6, 'replication-factor': 1},
+                                      self.output_topic_1: {'partitions': 1, 'replication-factor': 1},
+                                      self.output_topic_2: {'partitions': 1, 'replication-factor': 1}
+                                  })
+
+    def get_consumer(self, topic, num_messages):
+        return VerifiableConsumer(self.test_context,
+                                  1,
+                                  self.kafka,
+                                  topic,
+                                  "stream-broker-resilience-verify-consumer",
+                                  max_messages=num_messages)
+
+    def assert_consume(self, test_state, topic, num_messages=5):
+        consumer = self.get_consumer(topic, num_messages)
+        consumer.start()
+
+        wait_until(lambda: consumer.total_consumed() >= num_messages,
+                   timeout_sec=120,
+                   err_msg="At %s streams did not process messages in 60 seconds " % test_state)
+
+    @staticmethod
+    def get_configs(extra_configs=""):
+        # Consumer max.poll.interval > min(max.block.ms, ((retries + 1) * request.timeout)
+        consumer_poll_ms = "consumer.max.poll.interval.ms=50000"
+        retries_config = "producer.retries=2"
+        request_timeout = "producer.request.timeout.ms=15000"
+        max_block_ms = "producer.max.block.ms=30000"
+
+        # java code expects configs in key=value,key=value format
+        updated_configs = consumer_poll_ms + "," + retries_config + "," + request_timeout + "," + max_block_ms + extra_configs
+
+        return updated_configs
+
+    def wait_for_verification(self, processor, message, file, num_lines=1, timeout_sec=20):
+        wait_until(lambda: self.verify_from_file(processor, message, file) >= num_lines,
+                   timeout_sec=timeout_sec,
+                   err_msg="Did expect to read '%s' from %s" % (message, processor.node.account))
+
+    @staticmethod
+    def verify_from_file(processor, message, file):
+        result = processor.node.account.ssh_output("grep '%s' %s | wc -l" % (message, file), allow_fail=False)
+        return int(result)
+
+
+    def setUp(self):
+        self.zk.start()
+        self.kafka.start()
+
+    def test_standby_tasks_rebalance(self):
+
+        driver = StreamsStandbyTaskDriverService(self.test_context, self.kafka, self.num_messages)
```

Comment:
Should we verify this driver is closed (i.e. specified num_messages have been sent) in the end before verification?

Reply:
ack